File: BubbleSort.rs

fn BubbleSort(arr : &mut [i64]){
  let n = arr.len();
  for i in 0..n-1{
    for j in 0..n-i-1{
      if arr[j] > arr[j+1]{
        let mut swap = arr[j];
        arr[j] = arr[j+1];
        arr[j+1] = swap;
      }
    }
  }
}


fn main(){
  let mut arr = [23,45,9,1,2,0];
  BubbleSort(&mut arr);
  println!("Sort Array : {:?}",arr);
}

File: Bubblesort.php
<?php 

function BubbleSort($arr){
  $n = sizeof($arr);
  for ($i = 0; $i<$n-1; $i++){
    for ($j = 0; $j < $n-$i-1; $j++){
      if ($arr[$j] > $arr[$j+1]){
        $swap = $arr[$j];
        $arr[$j] = $arr[$j+1];
        $arr[$j+1] = $swap;
      }
    }
  }
  return $arr;
  
  echo $n ;
}


$arr = array(78,9,2,19,0,1);
$sortarr = BubbleSort($arr);
// print_r ($arr);
print_r ($sortarr);

?>
